src/serverModel.ts

```diff
@@ -153,28 +305,37 @@ export default function ServerModel(mpInstance) {
             };
         }
 
-        return jsonObject;
+        return jsonObject as IConsentStateDTO;
     };
 
-    this.createEventObject = function(event, user) {
-        var uploadObject = {};
-        var eventObject = {};
+    this.createEventObject = function(
+        event: BaseEvent,
+        user?: MParticleUser
+    ): SDKEvent | IUploadObject {
+        var uploadObject: Partial<IUploadObject> = {};
+        var eventObject: Partial<SDKEvent> = {};
 
+        // TODO: What is the purpose of setting OptOut to the inverse of the Store's
+        //       isEnabled property and not just `false`?
```

Comment:
```suggestion
        //  The `optOut` variable is passed later in this method to the `uploadObject`
        //  so that it can be used to denote whether or not a user has "opted out" of being
        //  tracked. If this is an `optOut` Event, we set `optOut` to the inverse of the SDK's 
        // `isEnabled` boolean which is controlled via `MPInstanceManager.setOptOut`.
```